builtin/providers/google/resource_container_cluster_test.go

```diff
@@ -135,53 +133,124 @@ func testAccCheckContainerClusterDestroy(s *terraform.State) error {
 	return nil
 }
 
-func testAccCheckContainerClusterExists(n string) resource.TestCheckFunc {
+func testAccCheckContainerCluster(n string) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
-		rs, ok := s.RootModule().Resources[n]
-		if !ok {
-			return fmt.Errorf("Not found: %s", n)
-		}
-
-		if rs.Primary.ID == "" {
-			return fmt.Errorf("No ID is set")
+		attributes, err := getResourceAttributes(n, s)
+		if err != nil {
+			return err
 		}
 
 		config := testAccProvider.Meta().(*Config)
-
-		attributes := rs.Primary.Attributes
-		found, err := config.clientContainer.Projects.Zones.Clusters.Get(
+		cluster, err := config.clientContainer.Projects.Zones.Clusters.Get(
 			config.Project, attributes["zone"], attributes["name"]).Do()
 		if err != nil {
 			return err
 		}
 
-		if found.Name != attributes["name"] {
-			return fmt.Errorf("Cluster not found")
+		if cluster.Name != attributes["name"] {
+			return fmt.Errorf("Cluster %s not found, found %s instead", attributes["name"], cluster.Name)
 		}
 
-		return nil
-	}
-}
+		type clusterTestField struct {
+			tf_attr  string
+			gcp_attr interface{}
+		}
 
-func testAccCheckContainerClusterAdditionalZonesExist(n string, num int) resource.TestCheckFunc {
-	return func(s *terraform.State) error {
-		rs, ok := s.RootModule().Resources[n]
-		if !ok {
-			return fmt.Errorf("Not found: %s", n)
+		clusterTests := []clusterTestField{
+			{"initial_node_count", strconv.FormatInt(cluster.InitialNodeCount, 10)},
+			{"master_auth.0.client_certificate", cluster.MasterAuth.ClientCertificate},
+			{"master_auth.0.client_key", cluster.MasterAuth.ClientKey},
+			{"master_auth.0.cluster_ca_certificate", cluster.MasterAuth.ClusterCaCertificate},
+			{"master_auth.0.password", cluster.MasterAuth.Password},
+			{"master_auth.0.username", cluster.MasterAuth.Username},
+			{"zone", cluster.Zone},
+			{"cluster_ipv4_cidr", cluster.ClusterIpv4Cidr},
+			{"description", cluster.Description},
+			{"endpoint", cluster.Endpoint},
+			{"instance_group_urls", cluster.InstanceGroupUrls},
+			{"logging_service", cluster.LoggingService},
+			{"monitoring_service", cluster.MonitoringService},
+			// TODO(danawillow): Add this back in. Currently this field is saved via the config instead of from the API response,
+			// and the config may contain the network name or self_link, whereas the API only returns the self_link.
+			// {"network", cluster.Network},
```

Comment:
Is there a way to derive the network name from the self_link? If so, could we just normalize on it in the test (i.e., convert both to the network name) so we could actually test it?